src/middlewared/middlewared/plugins/network.py

```diff
@@ -461,7 +461,9 @@ def sync(self):
         else:
             gc = self.middleware.call('datastore.query', 'network.globalconfiguration', None, {'get': True})
             if gc['gc_domain']:
-                domain = gc['gc_domain']
+                domains += gc['gc_domain']
```

Comment:
gc['gc_domain'] si not a list, so you cannot do +=
`domains.append(gc['gc_domain'])`